### PR TITLE
build: Simplify import_filename handling

### PR DESCRIPTION
**Pull request by xclaesse**
This removes deadcode, vs_import_filename and gcc_import_filename were not needed.

### CODE REVIEW

File: mesonbuild/build.py [hunk 1]
```diff
@@ -2083,6 +2072,9 @@ def post_init(self) -> None:
         # libfoo.a. However, we cannot use foo.lib because that's the same as
         # the import library. Using libfoo.a is ok because people using MSVC
         # always pass the library filename while linking anyway.
+        #
+        # See our FAQ for more detailed ratinal:
```
Comment (dnicolodi) [inline mesonbuild/build.py:2076]: typo: `s/ratinal/rational/`

Comment (xclaesse) [reply mesonbuild/build.py]: fixed

**Comment by xclaesse** [comment]
I've seen the cygwin error in other PRs unfortunately, seems unrelated.

File: mesonbuild/build.py [hunk 2]
```diff
@@ -2083,6 +2072,9 @@ def post_init(self) -> None:
         # libfoo.a. However, we cannot use foo.lib because that's the same as
         # the import library. Using libfoo.a is ok because people using MSVC
         # always pass the library filename while linking anyway.
+        #
+        # See our FAQ for more detailed rational:
```
Comment (tristan957) [inline mesonbuild/build.py:2076]: s/rational/rationale

Comment (xclaesse) [reply mesonbuild/build.py]: TIL: 'rational' is an adjective and 'rationale' is a noun. Fixed.

**Comment by eli-schwartz** [comment]
(I haven't looked at the PR itself yet, but this caught my attention.)

> I've seen the cygwin error in other PRs unfortunately, seems unrelated.

Debugged and reported as https://gitlab.kitware.com/cmake/cmake/-/issues/25200

zlib 1.3 changed an innocuous header define and broke cmake's heuristic regex-based approach (it's a regex that implements `str.split('.')` which is an amazing thought!!!) to ad-hoc NIH reimplementing pkg-config. And cygwin has updated their zlib package (cc @jon-turney) which triggers the cmake bad behavior.

I'm disappointed that cmake decided to deny that regex is a problem, switch my bug report title, and move on to discussing the best form of regex to use, but hardly surprised... cmake relies quite heavily on regex... anyway, there's a pending hack to make that regex slightly less fragile. So the issue should be papered over soon.

**Comment by xclaesse** [comment]
Thanks for investigating @eli-schwartz !

**Comment by jon-turney** [comment]
Thanks for making me aware of that.  I'll see what I can do to get cmake updated...

(I wish I could give you a more stable set of packages to run CI against, but we don't really have good mechanisms for that currently)

**Comment by eli-schwartz** [comment]
Yeah, I guess it would be nice if there was some easy way to do what we currently do with docker to freeze Linux packages into a container only when CI is currently passing, except for Windows. I really have no idea what the Windows story is there, other than apparently "complicated".

**Comment by jon-turney** [comment]
> Yeah, I guess it would be nice if there was some easy way to do what we currently do with docker to freeze Linux packages into a container only when CI is currently passing, except for Windows. I really have no idea what the Windows story is there, other than apparently "complicated".

Last time I looked at this (which was a few years ago) fetching a Windows Docker container into a GH actions VM took even longer than just installing Cygwin (possibly due to the size of the base installation or something; using the nanoserver base image might help with that, but pulling on that thread uncovered other problems...)

As an alternative, maybe #12151 is a possible approach.

**Comment by xclaesse** [comment]
Back to original PR, anyone wants to give a review please?

**Review by tristan957 — APPROVED** [review]

**Pull request merged**